### DISCUSSION TITLE
How to set the zoom and position to make the graphics appear to fill the canvas?

### DISCUSSION THREAD

**Discussion by hungtcs** [klinecharts/KLineChart#504] [Q&A]
When there is very few data, the graphs are all concentrated on the far right.
Is there a way to zoom the graphic to the entire canvas via the instance api?

Default effect:

<img width="1621" alt="image" src="https://github.com/liihuu/KLineChart/assets/11692935/be84691f-beb9-4186-9e45-7ce849ae94ae">

I hope it can be adjusted through the instance api:

<img width="1623" alt="image" src="https://github.com/liihuu/KLineChart/assets/11692935/86341219-b882-43dd-a831-dbf64120f918">

Thanks.

**Comment by char101** [comment]
One way is to iteratively zoom until the left offset is some specified value

```javascript
const tss = chart._chartStore.getTimeScaleStore();
while (chart.getVisibleRange().realFrom < -5) {
  tss.setBarSpace(tss.getBarSpace().bar + 1);
}
```
